nodejs/src/index.js

```diff
@@ -7,34 +7,43 @@ var Compression = exports.Compression = {
 	Gzip: 1
 };
 
-exports.write = function (struct, compression, callback) {
-	switch (compression || Compression.None) {
-		case Compression.None:
-			serialize(struct, callback);
-			break;
-		case Compression.Gzip:
-			zip(struct, callback);
-			break;
-		default:
-			callback(new Error('Unable to understand the compression applied'));
-			break;
-	}
+exports.write = function (struct, compression, callback, noSettings) {
+    if (!noSettings) {
```

Comment:
Ok, I've now changed this to checking the number of arguments passed to the function.